pkg/apis/crd/v1alpha1/types.go

```diff
@@ -427,3 +354,132 @@ type BGPPeer struct {
 	// a restart before deleting stale routes. The range of the value is from 1 to 3600, and the default value is 120.
 	GracefulRestartTimeSeconds *int32 `json:"gracefulRestartTimeSeconds,omitempty"`
 }
+
+// Source describes the source spec of the packetcapture.
+type Source struct {
+	// Namespace is the source namespace.
+	Namespace string `json:"namespace"`
+	// Pod is the source pod.
+	Pod string `json:"pod"`
+	// IP is the source IPv4 or IPv6 address.
+	IP *string `json:"ip,omitempty"`
```

Comment:
you can remove all the string pointers (replace `*string` with `string`).
We had a long discussion about the use of pointers during the `BGPPolicy` API review: https://github.com/antrea-io/antrea/pull/6009#discussion_r1614020224
At least for strings which are introduced as API fields from the get-go(i.e., not added later on), I would avoid pointers. AFAIK we don't have a case where we need to distinguish been "unset" and "empty" string. Here "empty" will always mean "omitted".

Reply:
OK!